utils/visualizer/qviz/cube.py

```diff
@@ -5,7 +5,14 @@
 
 
 class Cube:
-    def __init__(self, cube_string: str, max_weight: int, element_count: int, size: int, depth: int):
+    def __init__(
+        self,
+        cube_string: str,
+        max_weight: int,
+        element_count: int,
+        size: int,
```

Comment:
You may need to modify this class because different blocks can belong to the same file.
Different blocks can belong to the same file. The `size` here will cause incorrect `sampled_size` and `total_size` in `SamplingInfo.`

Reply:
Why would it cause an incorrect sampled_size and total_size?

Right now when we extract the cubes from a delta table, we read each block and:

- **If we already have a cube with the block's cubeID** the size associated to that block is added to the cube's size.
- **If we don't have a cube with that cubeID** we create a new one and it's size is defined as the size associated to that block.

Reply:
I think you are right @jorgeMarin1 , if we update the cube parameters by processing each block separately, we are going to get the correct total size and sampled size. 

Let me just test it locally and see that the information is correct 👍